chalice/invoke.py

```diff
@@ -0,0 +1,88 @@
+"""Abstraction for invoking a lambda function."""
+import json
+
+from typing import Any, Optional, Dict, List, Union  # noqa
+
+from chalice.config import DeployedResources  # noqa
+from chalice.awsclient import TypedAWSClient  # noqa
+from chalice.utils import UI  # noqa
+
+_OPT_STR = Optional[str]
+
+
+class NoSuchFunctionError(Exception):
+    """The specified function could not be found."""
+    def __init__(self, name):
+        # type: (str) -> None
+        self.name = name
+        super(NoSuchFunctionError, self).__init__()
+
+
+class LambdaInvoker(object):
+    def __init__(self, deployed_resources, client):
+        # type: (DeployedResources, TypedAWSClient) -> None
+        self._deployed_resources = deployed_resources
+        self._client = client
+
+    def invoke(self, name, payload=None):
+        # type: (str, _OPT_STR) -> Dict[str, Any]
+        try:
+            resource = self._deployed_resources.resource_values(name)
+            lambda_arn = resource['lambda_arn']
+        except (ValueError, KeyError):
+            raise NoSuchFunctionError(name)
+
+        # TODO For now we just let the error propogate. It might be useful
+        # later to inspect the type of the resource we are invoking to know
+        # its event source and look at the error and know if we can safely
+        # retry.
+        return self._client.invoke_function(
+            lambda_arn,
+            payload=payload
+        )
+
+
+class LambdaResponseFormatter(object):
+    _ERROR_KEY = 'FunctionError'
+    _ERROR_VALUE = 'Unhandled'
+    _PAYLOAD_KEY = 'Payload'
+
+    _TRACEBACK_HEADING = 'Traceback (most recent call last):\n'
+
+    def __init__(self, ui):
+        # type: (UI) -> None
+        self._ui = ui
+
+    def format_response(self, response):
+        # type: (Dict[str, Any]) -> None
+        payload = response[self._PAYLOAD_KEY].read()
+        if self._ERROR_KEY in response and \
```

Comment:
Can be refactored to:

`if response.get(self._ERROR_KEY) == self._ERROR_VALUE`